lib/link.js

```diff
@@ -26,6 +26,12 @@ export default class Link extends Component {
     const route = as ? href : null
     const url = as || href
 
+    //  avoid scroll for urls with anchor refs
+    let { scroll } = this.props
+    if (url.indexOf('#') > -1 && !scroll) {
```

Comment:
Thanks @nkzawa ! I do have one concern though. The scroll check is explicitly looking for `false`:
```
if (scroll !== false) window.scrollTo(0, 0)
```

If no scroll prop is passed in then it wouldn't pass the condition `scroll != null` and would skip the check for `url.indexOf('#') < 0`. Which I believe is the scenario we wanted to cover in this fix.